### DISCUSSION TITLE
Is it possible to run npm programs that run local scripts which require from a local node modules folder?

### DISCUSSION THREAD

**Discussion by schuelermine** [denoland/deno#23256] [Q&A] [answered]
For instance, [Eleventy](11ty.dev/) executes a script `.eleventy.js` which, when using Node, can call `require` to import packages from `node_modules`.

But when using `deno run --node-modules-dir npm:@11ty/eleventy` this fails.

Curiously, it does succeed when `node_modules` is generated by npm, not Deno, but Eleventy is still run with Deno.

I can't convert the imports to ES modules because Eleventy itself `require`s the file, so it causes an error.

Why can the file not import from the node modules directory? Is there some workaround?

**Comment by bartlomieju** [comment]
Can you paste what's the exact error?

**Reply by schuelermine** [reply, marked as answer]
I figured out what the problem was. I was importing an indirect dependency, which works if node_modules is laid out like npm does it, but not like Deno does it.
